### PR TITLE
HTTP/3 support by default in .NET 8

### LINKED ISSUE #28598: HTTP/3 enabled by default for Kestrel
https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-1/#http-3-enabled-by-default

related  #28600

See https://github.com/dotnet/AspNetCore.Docs/issues/28599#issuecomment-1477035590

cc @tdykstra  if you're interested.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 57282931-0741-9ae0-1bc3-1918160769e2
* Version Independent ID: d8ff0551-6287-7921-b18d-2cad70fce012
* Content: [Use HTTP/3 with the ASP.NET Core Kestrel web server](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/http3?view=aspnetcore-7.0)
* Content Source: [aspnetcore/fundamentals/servers/kestrel/http3.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/servers/kestrel/http3.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-fundamentals**
* GitHub Login: @wtgodbe
* Microsoft Alias: **wigodbe**

---
[Associated WorkItem - 68934](https://dev.azure.com/msft-skilling/Content/_workitems/edit/68934)

### PR DESCRIPTION
**Pull request by tdykstra**
Fixes #28598 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/fundamentals/servers/kestrel.md](https://github.com/dotnet/AspNetCore.Docs/blob/abf7eb3a25fbf42e9544e225a004f9cd5ea85c9a/aspnetcore/fundamentals/servers/kestrel.md) | [aspnetcore/fundamentals/servers/kestrel](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel?branch=pr-en-us-28876) |
| [aspnetcore/fundamentals/servers/kestrel/endpoints.md](https://github.com/dotnet/AspNetCore.Docs/blob/abf7eb3a25fbf42e9544e225a004f9cd5ea85c9a/aspnetcore/fundamentals/servers/kestrel/endpoints.md) | [aspnetcore/fundamentals/servers/kestrel/endpoints](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/endpoints?branch=pr-en-us-28876) |
| [aspnetcore/fundamentals/servers/kestrel/http3.md](https://github.com/dotnet/AspNetCore.Docs/blob/abf7eb3a25fbf42e9544e225a004f9cd5ea85c9a/aspnetcore/fundamentals/servers/kestrel/http3.md) | [aspnetcore/fundamentals/servers/kestrel/http3](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/servers/kestrel/http3?branch=pr-en-us-28876) |
| [aspnetcore/grpc/troubleshoot.md](https://github.com/dotnet/AspNetCore.Docs/blob/abf7eb3a25fbf42e9544e225a004f9cd5ea85c9a/aspnetcore/grpc/troubleshoot.md) | [aspnetcore/grpc/troubleshoot](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/troubleshoot?branch=pr-en-us-28876) |

### CODE REVIEW

File: aspnetcore/fundamentals/servers/kestrel/http3.md [hunk 1]
```diff
@@ -48,22 +44,13 @@ HTTP/3 isn't currently supported on macOS and may be available in a future relea
...
-
-Because not all routers, firewalls, and proxies properly support HTTP/3, HTTP/3 should be configured together with HTTP/1.1 and HTTP/2. This can be done by specifying [`HttpProtocols.Http1AndHttp2AndHttp3`](xref:Microsoft.AspNetCore.Server.Kestrel.Core.HttpProtocols.Http1AndHttp2AndHttp3) as an endpoint's supported protocols.
+HTTP/3 is enabled by default. Because not all routers, firewalls, and proxies properly support HTTP/3, by default HTTP/3 is configured together with HTTP/1.1 and HTTP/2.
 
-For more information, see <xref:fundamentals/servers/kestrel/endpoints>.
+For more information, see <xref:fundamentals/servers/kestrel/endpoints#listenoptionsprotocols>.
 
 ## Alt-svc
 
-HTTP/3 is discovered as an upgrade from HTTP/1.1 or HTTP/2 via the [`alt-svc`](https://developer.mozilla.org/docs/Web/HTTP/Headers/Alt-Svc) header. That means the first request will normally use HTTP/1.1 or HTTP/2 before switching to HTTP/3. Kestrel automatically adds the `alt-svc` header if HTTP/3 is enabled.
```
Comment (Tratcher) [inline aspnetcore/fundamentals/servers/kestrel/http3.md:66]: "if HTTP/3 is enabled" is still relevant. If you turn off HTTP/3 then alt-svc is also turned off.

File: aspnetcore/grpc/troubleshoot.md [hunk 2]
```diff
@@ -232,9 +228,9 @@ Alternatively, a client factory can be configured with `SubdirectoryHandler` by
 
 ## Configure gRPC client to use HTTP/3
 
-The .NET gRPC client supports HTTP/3 with .NET 6 or later. If the server sends an `alt-svc` response header to the client that indicates the server supports HTTP/3, the client will automatically upgrade its connection to HTTP/3. For information about how to enable HTTP/3 on the server, see <xref:fundamentals/servers/kestrel/http3>.
+The .NET gRPC client supports HTTP/3 with .NET 6 or later. If the server sends an `alt-svc` response header to the client that indicates the server supports HTTP/3, the client will automatically upgrade its connection to HTTP/3. The Kestrel server supports HTTP/3 by default. For more information, see <xref:fundamentals/servers/kestrel/http3>.
 
-HTTP/3 support is in preview in .NET 6, and needs to be enabled via a configuration flag in the project file:
+HTTP/3 support is in preview in .NET 8, and needs to be enabled via a configuration flag in the project file:
```
Comment (Tratcher) [inline aspnetcore/grpc/troubleshoot.md:233]: @JamesNK isn't this obsolete?

File: aspnetcore/fundamentals/servers/kestrel/endpoints.md [hunk 3]
```diff
@@ -11,6 +11,8 @@ uid: fundamentals/servers/kestrel/endpoints
 
 # Configure endpoints for the ASP.NET Core Kestrel web server
 
+[!INCLUDE[](~/blazor/includes/not-latest-version.md)]
```
Comment (Tratcher) [inline aspnetcore/fundamentals/servers/kestrel/endpoints.md:14]: ?

File: aspnetcore/fundamentals/servers/kestrel/http3.md [hunk 4]
```diff
@@ -48,22 +44,13 @@ HTTP/3 isn't currently supported on macOS and may be available in a future relea
...
+HTTP/3 is discovered as an upgrade from HTTP/1.1 or HTTP/2 via the [`alt-svc`](https://developer.mozilla.org/docs/Web/HTTP/Headers/Alt-Svc) header. That means the first request will normally use HTTP/1.1 or HTTP/2 before switching to HTTP/3. Kestrel automatically adds the `alt-svc` header.
```
Comment (tdykstra) [inline aspnetcore/fundamentals/servers/kestrel/http3.md:53]:
```suggestion
HTTP/3 is discovered as an upgrade from HTTP/1.1 or HTTP/2 via the [`alt-svc`](https://developer.mozilla.org/docs/Web/HTTP/Headers/Alt-Svc) header. That means the first request will normally use HTTP/1.1 or HTTP/2 before switching to HTTP/3. Kestrel automatically adds the `alt-svc` header if HTTP/3 is enabled.
```

**Review by JamesNK — APPROVED** [review]

**Pull request merged**